### PR TITLE
Suisin/scam message warning for brazil binary

### PR DESCRIPTION
**Pull request by suisin-deriv**

### CODE REVIEW

File: src/javascript/app/base/page.js [hunk 1]
```diff
@@ -133,13 +134,16 @@ const Page = (() => {
                 }, 1000);
                 RealityCheck.onLoad();
                 Menu.init();
+                const is_at_brazil = State.getResponse('website_status.clients_country') === 'br';
```
Comment (mahdiyeh-fs) [inline src/javascript/app/base/page.js:137]: shouldn't it be like this:
`                const is_at_brazil = (Client.get('residence') === 'br' ||State.getResponse('website_status.clients_country') === 'br');
`
also better to name it `is_br_client`

File: src/javascript/app/base/client.js [hunk 2]
```diff
@@ -120,6 +120,9 @@ const Client = (() => {
         if (response.logout !== 1) return;
         removeCookies('login', 'loginid', 'loginid_list', 'email', 'residence', 'settings'); // backward compatibility
         removeCookies('reality_check', 'affiliate_token', 'affiliate_tracking', 'onfido_token', 'hide_guide');
+        // clear local storage
+        localStorage.removeItem('readScamMessage');
+        localStorage.removeItem('isNewAccount');
```
Comment (mahdiyeh-fs) [inline src/javascript/app/base/client.js:125]:
```suggestion
        localStorage.removeItem('is_new_account');
```

File: src/javascript/app/base/page.js [hunk 3]
```diff
@@ -27,6 +27,7 @@ const Url              = require('../../_common/url');
 const createElement    = require('../../_common/utility').createElement;
 const isLoginPages     = require('../../_common/utility').isLoginPages;
 const isProduction     = require('../../config').isProduction;
+const WarningScamMessage = require('../pages/user/warning_scam_message');
```
Comment (mahdiyeh-fs) [inline src/javascript/app/base/page.js:30]: The `=` signs should align vertically due to the binary code base

File: src/javascript/app/base/page.js (hunk 1 above)
Comment (suisin-deriv) [reply src/javascript/app/base/page.js:137]: I just wanted client which they access their account in brazil.

File: src/javascript/app/base/client.js [hunk 4]
```diff
@@ -120,6 +120,9 @@ const Client = (() => {
         if (response.logout !== 1) return;
         removeCookies('login', 'loginid', 'loginid_list', 'email', 'residence', 'settings'); // backward compatibility
         removeCookies('reality_check', 'affiliate_token', 'affiliate_tracking', 'onfido_token', 'hide_guide');
+        // clear local storage
+        localStorage.removeItem('read_scam_message');
+        localStorage.removeItem('new_user_account');
```
Comment (matin-binary) [inline src/javascript/app/base/client.js:125]: You can add these under `client.accounts` and let `ClientBase.clearAllAccounts();` remove it all.

File: src/javascript/app/pages/user/warning_scam_message.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    if (actionParams === 'signup') {
+        localStorage.setItem('new_user_account', true);
+    }
+
+    const has_read_warning_scam_message = () => {
+        BinarySocket.wait('authorize', 'website_status').then(()=> {
+            const is_logged_in = !!Client.isLoggedIn();
+            const is_brazil = State.getResponse('website_status.clients_country') === 'br';
+            const is_message_read = localStorage.getItem('read_scam_message') || false;
+            const is_new_account = localStorage.getItem('new_user_account') || false;
```
Comment (yashim-deriv) [inline src/javascript/app/pages/user/warning_scam_message.js:20]:
```suggestion
            const is_message_read = !!localStorage.getItem('read_scam_message');
            const is_new_account = !!localStorage.getItem('new_user_account');
```
Doing `!!` will coerce null values to false

File: src/templates/app/user/warning_scam_message.jsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    <React.Fragment>
+        <div id='warning_scam_message' className='lightbox invisible'>
+            <div id='warning_scam_message_content'>
+                <h1 id='warning_scam_message_title'>{it.L('Beware of the fake links.')}</h1>
+                <p id='warning_scam_message_subtitle'>{it.L('A link can contain the word "Binary" and still be fake.')}</p>
+                <h2 id='warning_scam_message_title_secondary'>{it.L('Do not get lured to fake "Binary" pages!')}</h2>
+                <p id='warning_scam_message_text_secondary'>{it.L('You may see links to websites with a fake Binary login page where you’ll get scammed for your money. ')}</p>
+                <div className='warning-scam-message-info-message'>
+                    <img className='icon-sm margin-right-0' src={it.url_for('images/pages/warning_scam_message/ic-account-cross.svg')} />
+                    <h3 id='warning_scam_message_scam_title_text'>{it.L('Do not trust and give away your credentials on fake websites, ads or emails.')}</h3>
```
Comment (yashim-deriv) [inline src/templates/app/user/warning_scam_message.jsx:13]:
```suggestion
        <div id='warning_scam_message' className='lightbox invisible'>
            <div id='warning_scam_message_content'>
                <h1 id='warning_scam_message_title'>{it.L('Beware of the fake links.')}</h1>
                <p id='warning_scam_message_subtitle'>{it.L('A link can contain the word "Binary" and still be fake.')}</p>
                <h2 id='warning_scam_message_title_secondary'>{it.L('Do not get lured to fake "Binary" pages!')}</h2>
                <p id='warning_scam_message_text_secondary'>{it.L('You may see links to websites with a fake Binary login page where you’ll get scammed for your money. ')}</p>
                <div className='warning-scam-message-info-message'>
                    <img className='icon-sm margin-right-0' src={it.url_for('images/pages/warning_scam_message/ic-account-cross.svg')} />
                    <h3 id='warning_scam_message_scam_title_text'>{it.L('Do not trust and give away your credentials on fake websites, ads or emails.')}</h3>
```
Best not to use `ids` if there's no intention of changing the content dynamically from JavaScript

**Review by mahdiyeh-fs — DISMISSED** [review]

**Review by matin-binary — APPROVED** [review]

**Review by yashim-deriv — DISMISSED** [review]

**Pull request merged**